js/views/modals/NsfwWarning.js

```diff
@@ -0,0 +1,63 @@
+import $ from 'jquery';
+import app from '../../app';
+import loadTemplate from '../../utils/loadTemplate';
+import BaseModal from '../modals/BaseModal';
+
+export default class extends BaseModal {
+  constructor(options = {}) {
+    const opts = {
+      removeOnClose: true,
+      showCloseButton: false,
+      dismissOnEscPress: false,
+      ...options,
+    };
+
+    super(opts);
+    this.listenTo(app.settings, 'change:showNsfw', this.onChangeNsfw);
+  }
+
+  className() {
+    return `${super.className()} modalScrollPage modalMedium`;
+  }
+
+  events() {
+    return {
+      'click .js-cancel': 'onCancelClick',
+      'click .js-btnProceed': 'onProceedClick',
+      ...super.events(),
+    };
+  }
+
+  onCancelClick() {
+    this.trigger('canceled');
+    this.close();
+  }
+
+  onChangeNsfw(md, showNsfw) {
+    if (showNsfw) this.close();
+  }
+
+  onProceedClick() {
+    if (this.getCachedEl('.js-checkboxNsfw').is(':checked')) {
+      this.stopListening(app.settings, null, this.onChangeNsfw);
+      app.settings.set('showNsfw', true);
+      $.ajax({
+        type: 'PATCH',
+        url: app.getServerUrl('ob/settings/'),
+        data: JSON.stringify({ showNsfw: true }),
+        dataType: 'json',
+      });
```

Comment:
I was being optimistic and it just doesn't seem worth it to me to handle an error here. All it means is that the client will have `nsfw: true` until the next time the user starts the app, in which case it will come back as `false` when the settings are fetched. Once the user notices that it's false and they want it true, they could try again.

Also, this is local call that doesn't involve any type of communication with other nodes, so it's quite unlikely it will fail.